### PR TITLE
Feature/paging

### PR DESCRIPTION
**Pull request by joseantoniocampillo**
**ISSUE**
#28

**SUMMARY**
Add Pagin in principal list, and propuse light changes for UI style.

**CHANGES MADE**

Add Paging support with Paging 3 library.
Add extra with a LoadStateAdapter.
Little change on colors in styles.

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -81,13 +82,13 @@ dependencies {
     androidTestImplementation 'androidx.test.espresso:espresso-core:3.3.0'
 
     // Retrofit
-    implementation 'com.squareup.okhttp3:logging-interceptor:4.9.0'
+    implementation 'com.squareup.okhttp3:logging-interceptor:5.0.0-alpha.2'
```
Comment (juancoob) [inline app/build.gradle]: @joseantoniocampillo Why we do need the alpha version of retrofit?

File: app/src/main/java/com/architectcoders/arquitectomarvel/ui/main/AdapterList.kt [hunk 2]
```diff
@@ -44,3 +52,42 @@ class AdapterList(private val listener: (Result, View) -> Unit) :
         }
     }
 }
+
+const val INICIAL = 36
```
Comment (juancoob) [inline app/src/main/java/com/architectcoders/arquitectomarvel/ui/main/AdapterList.kt]: @joseantoniocampillo can you change the variable name to "STARTING"?

File: app/src/main/java/com/architectcoders/arquitectomarvel/ui/main/AdapterList.kt [hunk 3]
```diff
@@ -44,3 +52,42 @@ class AdapterList(private val listener: (Result, View) -> Unit) :
...
+    override suspend fun load(
+        params: LoadParams<Int>
+    ): LoadResult<Int, Result> {
+        return try {
+            val offset = params.key ?: INICIAL
+            val response = repository.getCharactersRemote(offset)
+            val results = response.characterData?.results!!
+            LoadResult.Page(
+                data = results,
+//                prevKey = if (offset == INICIAL) null else offset + INICIAL,
```
Comment (juancoob) [inline app/src/main/java/com/architectcoders/arquitectomarvel/ui/main/AdapterList.kt]: @joseantoniocampillo is this line necessary?

File: app/src/main/java/com/architectcoders/arquitectomarvel/ui/main/MainActivity.kt [hunk 4]
```diff
@@ -1,41 +1,79 @@
...
+        val mLayoutManager = GridLayoutManager(this@MainActivity, 3)
+        val footerAdapter = ResultsLoadStateAdapter(adapter::retry)
+        binding.apply {
+            mainHeroList.autoFitColumnsForGridLayout(resources.getDimension(R.dimen.avatar_width))
+            mainHeroList.layoutManager = mLayoutManager
+            mainHeroList.adapter = adapter.withLoadStateFooter(footerAdapter)
+            mLayoutManager.spanSizeLookup = object : GridLayoutManager.SpanSizeLookup() {
+                override fun getSpanSize(position: Int): Int {
+                    return if (position == adapter.itemCount && footerAdapter.itemCount > 0) {
+                        3
```
Comment (juancoob) [inline app/src/main/java/com/architectcoders/arquitectomarvel/ui/main/MainActivity.kt]: @joseantoniocampillo could you explain why this is necessary?

File: app/src/main/java/com/architectcoders/arquitectomarvel/ui/main/MainViewModel.kt [hunk 5]
```diff
@@ -1,84 +1,33 @@
...
-            tmpResult.forEach {
-                val itemForLocalList = dao.getResultWithItemsComics(it.comicsCollectionURI)
-                localList.addAll(itemForLocalList)
-            }*/
-        }
-    }
+    val pager = Pager(
+        config = PagingConfig(
+            pageSize = 18,
+//            maxSize = 54,
```
Comment (juancoob) [inline app/src/main/java/com/architectcoders/arquitectomarvel/ui/main/MainViewModel.kt]: @joseantoniocampillo is this line needed? Should we store the numbers like "pageSize" on dimens.xml?

**Review by juancoob — CHANGES_REQUESTED** [review]
Thanks @joseantoniocampillo Please, check my comments and let me know what do you think.

**Review by juancoob — APPROVED** [review]
Perfect @joseantoniocampillo !!

**Review by jujoru — APPROVED** [review]
OK¡

**Review by JorgeGreek — APPROVED** [review]
perfect

**Pull request merged**